Doc/whatsnew/3.13.rst

```diff
@@ -94,6 +94,24 @@ Improved Error Messages
   variables. See also :ref:`using-on-controlling-color`.
   (Contributed by Pablo Galindo Salgado in :gh:`112730`.)
 
+* A common mistake is to write a script with the same name as a
+  standard library module. The interpreter now detects this and
+  displays a more helpful error message.
+  (Contributed by Shantanu Jain in :gh:`95754`.)
+
+  .. code-block:: shell-session
+
+    $ python random.py
+    Traceback (most recent call last):
+      File "/home/random.py", line 1, in <module>
+        import random; print(random.randint(5))
+        ^^^^^^^^^^^^^
+      File "/home/random.py", line 1, in <module>
+        import random; print(random.randint(5))
+                            ^^^^^^^^^^^^^^
+    AttributeError: module 'random' has no attribute 'randint' (most likely due to '/home/random.py' shadowing the standard library module named 'random')
```

Comment:
This is a really big improvement, but I worry that beginners (who are most likely to be stumbling upon this error) still might not understand what's meant by "shadowing" the standard library. I wonder if something like this might be better -- tell them exactly what to do to fix the problem?

```suggestion
    AttributeError: module 'random' has no attribute 'randint' (consider renaming '/home/random.py', which may be shadowing the standard library 'random' module)
```

Reply:
+1 for actionable error messages.

* https://developers.google.com/tech-writing/error-messages/show-fix?hl=en
* https://developers.google.com/chat/design/errors
* https://learn.microsoft.com/en-gb/windows/win32/uxguide/mess-error